pkg/systemd/services/services.jsx

```diff
@@ -279,6 +284,47 @@ class ServicesPage extends React.Component {
         return Object.keys(service_tabs_suffixes).includes(suffix);
     }
 
+    /* When the page is running in the background fetch only information about failed units
+     * in order to update the 'Page Status'. The whole listUnits is very expensive.
+     * We still need to maintain the 'unit_by_path' state object so that if we receive
+     * some signal we can normally parse it and update only the affected unit state
+     * instead of calling ListUnitsFiltered API call for every received signal which
+     * might have changed the failed units array
+     */
+    listFailedUnits() {
+        return systemd_manager.ListUnitsFiltered(["failed"])
+                .then(failed => {
+                    for (let i = 0; i < failed.length; i++) {
+                        const result = failed[i];
+                        const path = result[6];
+                        const unit_id = result[0];
+
+                        if (!this.isUnitHandled(unit_id))
+                            continue;
+
+                        if (!this.seenPaths.has(path)) {
+                            this.seenPaths.add(path);
+                            this.path_by_id[unit_id] = path;
+                        }
+
+                        const unit = Object.assign({}, this.state.unit_by_path[path]);
+                        unit.path = path;
+                        unit.Id = unit_id;
+                        unit.LoadState = result[2];
+                        unit.ActiveState = result[3];
+                        unit.SubState = result[4];
+                        this.updateComputedProperties(unit);
+                        this.setState(prevState => ({
+                            unit_by_path: {
+                                ...prevState.unit_by_path,
+                                [path]: unit,
+                            }
+                        }));
+                    }
+                    this.processFailedUnits();
+                });
```

Comment:
I think we should catch failures here, like in the regular `ListUnits()` call, otherwise we get a page crash.